pkg/controller/controllername/controllername.go

```diff
@@ -0,0 +1,47 @@
+package controllername
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	metricsNameDelimiter = "_"
+	loggerNameDelimiter  = "-"
+)
+
+type ControllerNamer interface {
+	MetricsName() string
+	LoggerName() string
+}
+
+// controllerName ex. {"My","Controller", "Name"} ->  MyControllerName
+type controllerName []string
+
+func NewControllerName(name []string) controllerName {
+	cn := make(controllerName, len(name))
+
+	for i, w := range name {
+		cn[i] = removeSpace(strings.ToLower(w))
```

Comment:
should we also remove everything except alphabetic characters?

Reply:
i think so. I updated this to only allow letters